lib/write-set-tokens.ts

```diff
@@ -0,0 +1,90 @@
+import { types, Set, Range, Char, SetTokens } from './types';
+import * as sets from './sets-lookup';
+
+/**
+ * Takes character code and returns character to be displayed in a set
+ * @param {number} charCode Character code of set element
+ * @param {boolean} isFirstChar True if character is the first character of the set
+ * @param {boolean} isLastChar True if character is the last character of the set
+ * @returns {string} The string for the sets character
+ */
+export function setChar(charCode: number, isFirstChar: boolean, isLastChar = false): string {
+  // We only need to negate first internal '^' if token.not is false
+  return isFirstChar && charCode === 94 ? '\\^' :
```

Comment:
No it's not necessary, just thought it would make the outputs more readable. Happy to remove this minimal escaping if you and @fent think that is better approach :)